core/src/main/scala/cats/data/Cokleisli.scala

```diff
@@ -107,18 +107,15 @@ private trait CokleisliArrow[F[_]] extends Arrow[Cokleisli[F, ?, ?]] with Coklei
   override def dimap[A, B, C, D](fab: Cokleisli[F, A, B])(f: C => A)(g: B => D): Cokleisli[F, C, D] =
     super[CokleisliProfunctor].dimap(fab)(f)(g)
 
-  override def split[A, B, C, D](f: Cokleisli[F, A, B], g: Cokleisli[F, C, D]): Cokleisli[F, (A, C), (B, D)] =
-    super[CokleisliSplit].split(f, g)
+  override def split[A1, B1, A2, B2](f: Cokleisli[F, A1, B1], g: Cokleisli[F, A2, B2]): Cokleisli[F, (A1, A2), (B1, B2)] =
+    Cokleisli(fac => f.run(F.map(fac)(_._1)) -> g.run(F.map(fac)(_._2)))
```

Comment:
Right.  I was saying that the `split` method is not directly tested in the commutativity law, so it's not tested for both `Kleisli` and `Cokleisli`. I agree that it's probably fine since the commutativity law is the only **relevant** law to it and that together with parametricity probably it is enough to ensure its correctness (although, as you pointed out, in the noncommutative case the correctness is a bit arbitrary). 
I don't feel strong about it but adding a doctest (or just a regular unit test) to provide some codecov coverage wouldn't hurt right? I mean right now you can replace the implementation of these two `split` impl with `???`s without breaking any test, which is a bit uneasy to me.